packages/pages/src/common/src/template/internal/types.ts

```diff
@@ -52,6 +52,8 @@ export interface TemplateModuleInternal<
 export interface TemplateConfigInternal {
   /** The name of the template feature. If not defined uses the template filename (without extension) */
   name: string;
+  /** Determines if hydration is allowed or not for webpages. Defaults to false if not set*/
+  hydration?: boolean;
```

Comment:
hydrate

Reply:
Also this shouldn't be optional in the internal type. The point of internal is to get the inferred values from the external type so that they are available when we use it in code.